Model/review.model.js

```diff
@@ -4,15 +4,29 @@ exports.selectReviews = (query) => {
   // ONE QUERY ASH
   const columnName = Object.keys(query)[0];
   const whereValue = query[columnName];
+  const validColumns = [
+    "social deduction",
+    "euro game",
+    "dexterity",
+    "bananas", //example for category with no reviews.
+  ];
+
   let queryStr =
     "SELECT reviews.*, COUNT(comment_id) AS comment_count FROM reviews LEFT JOIN comments ON comments.review_id = reviews.review_id";
   if (columnName) {
     queryStr += ` WHERE ${columnName} = $1`;
   }
+
   queryStr += " GROUP BY reviews.review_id;";
 
   if (whereValue) {
+    if (!validColumns.includes(whereValue)) {
+      return Promise.reject({ status: 400, msg: "Bad Request" });
+    }
     return db.query(queryStr, [whereValue]).then((results) => {
+      if (results.rows.length === 0) {
+        return Promise.reject({ status: 200, msg: "No Content" });
+      }
```

Comment:
if we have results.rows.length === 0 this could mean 2 things:
-the category does exist but there are no reviews
-the category doesn't exist

to find out which one it is, we could do another query to the database to see if the category exists.